src/bin/addr2line.rs

```diff
@@ -90,13 +90,7 @@ fn main() {
         if let Some((file, lineno, func)) = loc {
             if show_funcs {
                 use std::borrow::Cow;
-                println!("{}",
-                         func.map(|func| if do_demangle {
-                                 Cow::Owned(demangle(&*func).to_string())
-                             } else {
-                                 func
-                             })
-                             .unwrap_or(Cow::Borrowed("??")));
+                println!("{}", func.unwrap_or(Cow::Borrowed("??")));
```

Comment:
Out of curiosity, what's the rationale for moving demangling into the `addr2line` internals? I kind of liked the fact that it was entirely external to the `addr2line` lookup logic.